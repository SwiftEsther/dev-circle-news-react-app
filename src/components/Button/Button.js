import React from 'react';

const Button = ({ children, style, ...props } /* props */) => {
    // console.log('style is', style)
    // props.children -> delete || Add headline
    const buttonStyle = {
        padding: 15,
        border: 'none',
        cursor: 'pointer',
        backgroundColor: 'blue',
        color: 'white',
        ...style
    };
    return (
        <button style={buttonStyle} {...props}> 
            { children }
        </button>
    );
}

export default Button;
