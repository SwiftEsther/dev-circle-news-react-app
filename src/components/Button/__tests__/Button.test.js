import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '../Button';


describe('Button', () => {
    describe('Snapshot', () => {
        it('should create a basic button with default styles', () => {
            const component = renderer
                .create(<Button>Click me</Button>)
                .toJSON();
            expect(component).toMatchSnapshot();
        });
        it('Should create a button with custom styles', () => {
            const customButtonComponent = renderer
                .create(
                    <Button style={{ backgroundColor: 'red' }}>
                        anything
                    </Button>
                ).toJSON()
            expect(customButtonComponent).toMatchSnapshot();
        })
    });

    describe('Unit', () => {
        it('should create a clickable button', () => {
            let count = 0;
            const button = shallow(
                <Button onClick={() => count++}>
                    click me!
                </Button>
            );
            button.simulate('click');
            expect(count).toEqual(1);
        });
    });
})