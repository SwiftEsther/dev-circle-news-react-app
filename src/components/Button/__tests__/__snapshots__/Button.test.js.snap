// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Should create a button with custom styles 1`] = `
<button
  style={
    Object {
      "backgroundColor": "red",
      "border": "none",
      "color": "white",
      "cursor": "pointer",
      "padding": 15,
    }
  }
>
  anything
</button>
`;

exports[`Button Snapshot should create a basic button with default styles 1`] = `
<button
  style={
    Object {
      "backgroundColor": "blue",
      "border": "none",
      "color": "white",
      "cursor": "pointer",
      "padding": 15,
    }
  }
>
  Click me
</button>
`;
